src/libcollections/btree/map.rs

```diff
@@ -875,20 +874,30 @@ impl<K: Ord, V: Ord> Ord for BTreeMap<K, V> {
     }
 }
 
-#[stable]
-impl<K: Show, V: Show> Show for BTreeMap<K, V> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        try!(write!(f, "BTreeMap {{"));
-
-        for (i, (k, v)) in self.iter().enumerate() {
-            if i != 0 { try!(write!(f, ", ")); }
-            try!(write!(f, "{:?}: {:?}", *k, *v));
-        }
+macro_rules! fmt_btree_map {
+    ($($Trait:ident),*) => {
+        $(
+            impl<K: fmt::$Trait, V: fmt::$Trait> fmt::$Trait for BTreeMap<K, V> {
+                fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                    try!(write!(f, "BTreeMap {{"));
+
+                    for (i, (k, v)) in self.iter().enumerate() {
+                        if i != 0 { try!(write!(f, ", ")); }
+                        try!(fmt::$Trait::fmt(k, f));
```

Comment:
Ah sorry for being a little slow to get back to this, but yes we don't want to apply the same flags to each element. Otherwise if you request an "at most 30-length formatted string" you'll get an at-most-30-length string for each element (instead of the whole formatting)